### PR TITLE
cli/demo: make the secure mode controllable via COCKROACH_INSECURE

### PR DESCRIPTION
**Pull request by knz**
(Note that this is useful regardless of where we settle as to what the default should be.)

Release note (cli change): It is now possible to pre-configure
the secure mode of `cockroach demo` using the environment
variable `COCKROACH_INSECURE` like other client commands.

### CODE REVIEW

File: pkg/cli/flags.go [hunk 1]
```diff
@@ -621,20 +621,20 @@ func init() {
 	demoFlags := demoCmd.PersistentFlags()
 	// We add this command as a persistent flag so you can do stuff like
 	// ./cockroach demo movr --nodes=3.
-	IntFlag(demoFlags, &demoCtx.nodes, cliflags.DemoNodes, 1)
-	BoolFlag(demoFlags, &demoCtx.runWorkload, cliflags.RunDemoWorkload, false)
+	IntFlag(demoFlags, &demoCtx.nodes, cliflags.DemoNodes, demoCtx.nodes)
+	BoolFlag(demoFlags, &demoCtx.runWorkload, cliflags.RunDemoWorkload, demoCtx.runWorkload)
 	VarFlag(demoFlags, &demoCtx.localities, cliflags.DemoNodeLocality)
-	BoolFlag(demoFlags, &demoCtx.geoPartitionedReplicas, cliflags.DemoGeoPartitionedReplicas, false)
+	BoolFlag(demoFlags, &demoCtx.geoPartitionedReplicas, cliflags.DemoGeoPartitionedReplicas, demoCtx.geoPartitionedReplicas)
```
Comment (rohany) [inline pkg/cli/flags.go:627]: does this change allow all of these flags to be set with an environment variable default?

Comment (knz) [reply pkg/cli/flags.go:627]: If you add an EnvVar field on the struct in  `cliflags/flags.go`, yes.

**Comment by rohany** [comment]
LGTM. Sorry I didn't see the different commits and was confused about why some code was being changed.

**Comment by knz** [comment]
thank you!

bors r=rohany

**Comment by knz** [comment]
transient failure, retrying

bors r=rohany

**Pull request merged**